pkg/config/config.go

```diff
@@ -38,6 +38,106 @@ type GochanConfig struct {
+			TemplateDir:  "/var/www/html/templates",
+			LogDir:       "/var/log/gochan",
+			Plugins:      []string{},
+			PluginSettings: map[string]any{
+				"examplePlugin": true,
+			},
+			SiteHeaderURL: "http://example.com",
+			WebRoot:       "/",
+			SiteDomain:    "example.com",
+			Verbose:       true,
+			RandomSeed:    "testseed",
+		},
+		SiteConfig: SiteConfig{
+			FirstPage:             []string{"index"},
+			Username:              "admin",
+			CookieMaxAge:          "1h",
+			StaffSessionDuration:  "2h",
+			Lockdown:              false,
+			LockdownMessage:       "",
+			SiteName:              "Test Site",
+			SiteSlogan:            "A testing site",
+			Modboard:              "mod",
+			MaxRecentPosts:        10,
+			RecentPostsWithNoFile: false,
+			EnableAppeals:         true,
+			MinifyHTML:            false,
+			MinifyJS:              false,
+			GeoIPType:             "none",
+			GeoIPOptions:          make(map[string]any),
+			Captcha: CaptchaConfig{
+				Type:                 "none",
+				OnlyNeededForThreads: false,
+				SiteKey:              "",
+				AccountSecret:        "",
+			},
+			FingerprintVideoThumbnails: false,
+			FingerprintHashLength:      16,
+		},
+		BoardConfig: BoardConfig{
+			InheritGlobalStyles: true,
+			Styles:              []Style{},
+			DefaultStyle:        "default",
+			Banners:             []PageBanner{},
+			PostConfig: PostConfig{
+				MaxLineLength:            1000,
+				ReservedTrips:            []string{},
+				ThreadsPerPage:           10,
+				RepliesOnBoardPage:       5,
+				StickyRepliesOnBoardPage: 2,
+				NewThreadsRequireUpload:  false,
+				CyclicalThreadNumPosts:   100,
+				BanColors:                []string{"#FF0000"},
+				BanMessage:               "You are banned!",
+				EmbedWidth:               640,
+				EmbedHeight:              360,
+				EnableEmbeds:             true,
+				ImagesOpenNewTab:         false,
+				NewTabOnOutlinks:         true,
+				DisableBBcode:            false,
+			},
+			UploadConfig: UploadConfig{
+				RejectDuplicateImages: true,
+				ThumbWidth:            150,
+				ThumbHeight:           150,
+				ThumbWidthReply:       100,
+				ThumbHeightReply:      100,
+				ThumbWidthCatalog:     200,
+				ThumbHeightCatalog:    200,
+				AllowOtherExtensions:  map[string]string{"pdf": "application/pdf"},
+				StripImageMetadata:    "none",
+				ExiftoolPath:          "",
+			},
+			DateTimeFormat:         "2006-01-02 15:04:05",
+			ShowPosterID:           true,
+			EnableSpoileredImages:  false,
+			EnableSpoileredThreads: false,
+			Worksafe:               false,
+			ThreadPage:             1,
+			Cooldowns:              BoardCooldowns{NewThread: 60, Reply: 30, ImageReply: 15},
+			RenderURLsAsLinks:      true,
+			ThreadsPerPage:         10,
+			EnableGeoIP:            false,
+			EnableNoFlag:           false,
+			CustomFlags:            []geoip.Country{},
+			isGlobal:               true,
+		},
+		jsonLocation: "test_config.json",
+		testing:      true,
+	}
+}
```

Comment:
This function isn't really necessary. Calls to `config.SetVersion("4.0.1")` essentially do the same thing, and SetVersion uses `defaultGochanConfig` which is also used in normal  operation

Reply:
I just modify the code and use SetVersion but there is a problem :
![image](https://github.com/user-attachments/assets/8e4cf543-2b2d-42f6-bba0-0ab4b8573f49)

Error : 
![image](https://github.com/user-attachments/assets/7ea7fc0a-182e-44e0-b22d-6c1c3d2b0e58)

Reply:
Other tests seems affected to `funcs_test.go->TestFormatTimestampTmplFunc()`.
But some tests using `SetVersion()->PanicIfNotTest()` are OK.

So i don't really know what's going on, i'll try to check why some test are panicking and other no.

Reply:
That's strange that it's not happening every time. `testutil.PanicIfNotTest` uses an admittedly less than elegant workaround for allowing certain test helper functions that need to be accessible from outside the package, I'm unable to reproduce that error from your code. What OS are you running this in, with what Go version, and how are you running the test (if in the command line, what command and arguments, and from what working directory)?

Reply:
Hey,

Sorry i was taking a break for the week-end.

So i'm currently use : 

- Windows 10 22H2
- Go version go1.23.3 windows/amd64
- Running tests throught "Go Test Explorer" extension (ID: premparihar.gotestexplorer)

I can use also a Fedora 24 OS, i will try at the end of the day

Reply:
Oh, it looks like that part was my fault, it wasn't anticipating Windows filename extensions or path separators in the test environment. I pushed a fix to the master branch, so if you pull that into your fork, it should work.

Reply:
Ok, nice I just check the 2 latest push on master seems good !
I will try now, and I gave you feedback in few hours 😄